web/client-api/src/main/java/io/deephaven/web/client/api/subscription/TableSubscription.java

```diff
@@ -31,80 +24,41 @@
-     */
-    public static final String EVENT_UPDATED = "updated";
-
+public final class TableSubscription extends AbstractTableSubscription {
 
-    // column defs in this subscription
-    private JsArray<Column> columns;
-    // holder for data
-    private SubscriptionTableData data;
+    private final JsArray<Column> columns;
+    private final Double updateIntervalMs;
 
-    // table created for this subscription
-    private Promise<JsTable> copy;
-
-    // copy from the initially given table so we don't need to way
     @JsIgnore
     public TableSubscription(JsArray<Column> columns, JsTable existingTable, Double updateIntervalMs) {
-
-        copy = existingTable.copy(false).then(table -> new Promise<>((resolve, reject) -> {
-            table.state().onRunning(newState -> {
-                // TODO handle updateInterval core#188
-                table.internalSubscribe(columns, this);
-
-                resolve.onInvoke(table);
-            }, table::close);
-        }));
-
+        super(existingTable.state(), existingTable.getConnection());
         this.columns = columns;
-        Integer rowStyleColumn = existingTable.state().getRowFormatColumn() == null ? NO_ROW_FORMAT_COLUMN
-                : existingTable.state().getRowFormatColumn().getIndex();
-        this.data = new SubscriptionTableData(columns, rowStyleColumn, this);
-
+        this.updateIntervalMs = updateIntervalMs;
     }
 
-    // public void changeSubscription(JsArray<Column> columns) {
-    // copy.then(t ->{
-    // t.internalSubscribe(columns, this);
-    // return Promise.resolve(t);
-    // });
-    // this.columns = columns;
-    // }
-
-
-    @JsIgnore
-    public void handleSnapshot(TableSnapshot snapshot) {
-        data.handleSnapshot(snapshot);
-    }
-
-    @JsIgnore
-    public void handleDelta(DeltaUpdates delta) {
-        data.handleDelta(delta);
+    @Override
+    protected void sendFirstSubscriptionRequest() {
+        changeSubscription(columns, updateIntervalMs);
     }
 
     /**
-     * The columns that were subscribed to when this subscription was created
+     * Updates the subscription to use the given columns and update interval.
      * 
-     * @return {@link Column}
+     * @param columns the new columns to subscribe to
+     * @param updateIntervalMs the new update interval, or null/omit to use the default of one second
      */
+    public void changeSubscription(JsArray<Column> columns, @JsNullable Double updateIntervalMs) {
+        sendBarrageSubscriptionRequest(null, columns, updateIntervalMs, false);
+    }
+
     @JsProperty
+    @Override
     public JsArray<Column> getColumns() {
-        return columns;
+        return super.getColumns();
     }
 
-    /**
-     * Stops the subscription on the server.
-     */
+    @Override
     public void close() {
-        copy.then(table -> {
-            table.close();
-            return Promise.resolve(table);
-        });
+        super.close();
```

Comment:
Is this superfluous?

Reply:
Looks like it, probably used to have other calls there - removed.